writeaheadlog.go

```diff
@@ -188,6 +188,10 @@ func (w *WAL) recoverWal(data []byte) ([]Update, error) {
 	}
 
 	if recoveryState == recoveryStateClean {
+		if err := w.writeRecoveryState(recoveryStateUnclean); err != nil {
+			return nil, err
```

Comment:
We should probably be extending this error to have more information about where recovery failed.

Reply:
e.g. `errors.Extend(err, errors.New("unable to write wal recovery state")`